services/web/server/src/simcore_service_webserver/session/_cookie_storage.py

```diff
@@ -0,0 +1,72 @@
+"""
+Extends aiohttp_session.cookie_storage
+
+"""
+
+import logging
+import time
+
+import aiohttp_session
+from aiohttp import web
+from aiohttp_session.cookie_storage import EncryptedCookieStorage
+
+_logger = logging.getLogger(__name__)
+
+
+def _share_cookie_across_all_subdomains(
+    request: web.BaseRequest, params: aiohttp_session._CookieParams
+) -> aiohttp_session._CookieParams:
+    """
+    Shares cookie across all subdomains, by appending a dot (`.`) in front of the domain name
+    overwrite domain from `None` (browser sets `example.com`) to `.example.com`
+    """
+    if (host := request.url.host) and host is not None:
+        params["domain"] = f".{host.lstrip('.')}"
+
+    return params
+
+
+class SharedCookieEncryptedCookieStorage(EncryptedCookieStorage):
+    async def save_session(
+        self,
+        request: web.Request,
+        response: web.StreamResponse,
+        session: aiohttp_session.Session,
+    ) -> None:
+        # link response to originating request (allows to detect the orginal request url)
+        response._req = request  # pylint:disable=protected-access  # noqa: SLF001
+
+        await super().save_session(request, response, session)
+
+    def save_cookie(
+        self,
+        response: web.StreamResponse,
+        cookie_data: str,
+        *,
+        max_age: int | None = None,
+    ) -> None:
+
+        params = self._cookie_params.copy()
+        if request := response._req:  # pylint:disable=protected-access  # noqa: SLF001
+            params = _share_cookie_across_all_subdomains(request, params)
```

Comment:
This is wrong. It's supposed to fail if response._req is None. You are not allowed to go forward. You are not migrating to an equivalent implementation. Remove the `if` and use asserts.